src/zir_sema.zig

```diff
@@ -1207,7 +1225,22 @@ fn errorUnionPayloadPtr(mod: *Module, scope: *Scope, unwrap: *zir.Inst.UnOp, saf
 fn errorUnionCode(mod: *Module, scope: *Scope, unwrap: *zir.Inst.UnOp) InnerError!*Inst {
     const tracy = trace(@src());
     defer tracy.end();
-    return mod.fail(scope, unwrap.base.src, "TODO implement zir_sema.errorUnionCode", .{});
+
+    const operand = try resolveInst(mod, scope, unwrap.positionals.operand);
+    if (operand.ty.zigTypeTag() != .ErrorUnion)
+        return mod.fail(scope, unwrap.base.src, "expected error union type, found '{}'", .{operand.ty});
+
+    if (operand.value()) |val| {
+        assert(val.getError() != null);
```

Comment:
Why have an assert here but a compiler error in `errorUnionPayload`? Also both of these need to be updated to handle pointers now that `.err_union_payload_unsafe_ptr` and friends have been added.

Reply:
There is an assert here because errorUnionCode is only generated by the compiler when it knows that there *is* an error. It cannot be called by the user, unlike errorUnionPayload. Does this make sense?

Reply:
> Also both of these need to be updated to handle pointers now that `.err_union_payload_unsafe_ptr` and friends have been added.

I just rebased without any issues. What do I need to update?

Reply:
It seems that `errorUnionPayloadPtr` and `errorUnionCodePtr` were added instead of calling the same functions. You should probably implement those too.

Reply:
I did it, but I don't really know how to test it.

Reply:
For `errorUnionPayloadPtr` you just need to use address of on a catch expression like:
```zig
const a: anyerror!u32 = 0;
const b: *u32 = &(a catch unreachable);
```

`errorUnionCodePtr` probably shouldn't even exist since `catch` doesn't allow pointer payloads, not sure why it was ever added.

Reply:
`errorUnionCodePtr` definetly shouldn't exist:
```zig
const a: anyerror!u32 = error.
const b = &(a catch |err| @compileLog(@TypeOf(err)));
```
```
stage2:
Compile Log Output:
@as(type, *const anyerror)

stage1:
| anyerror
```

Reply:
Oh yeah, I just can't read :). https://github.com/ziglang/zig/blob/master/src/zir_sema.zig#L1224